program/src/state.rs

```diff
@@ -134,20 +150,33 @@ pub struct Config {
 #[repr(C)]
 pub struct Proposal {
     discriminator: [u8; 8],
+    // Jon: it might be useful to include the governance that this is tied to
     /// The proposal author.
     pub author: Pubkey,
     /// Timestamp for when the cooldown period began.
     ///
     /// A `None` value means cooldown has not begun.
     pub cooldown_timestamp: Option<NonZeroU64>,
     /// Timestamp for when proposal was created.
+    // Jon: how about just using `UnixTimestamp` directly?
     pub creation_timestamp: u64,
+    // Jon: we should also have the voting start timestamp
     /// The instruction to execute, pending proposal acceptance.
+    // Jon: we can create a new type for the serialized instruction, but yeah, we'll
+    // need program id, account metas, instruction data, and then to sign with the
+    // governance PDA
     pub instruction: u64, // TODO: Replace with an actual serialized instruction?
     /// Amount of stake against the proposal.
     pub stake_against: u64,
     /// Amount of stake in favor of the proposal.
     pub stake_for: u64,
+    // Jon: it might be worth adding abstentions too, since that'll probably become a
+    // requested feature
+    // Jon: we'll need a bit more state machine management for proposals. To start,
+    // let's add a "status", ie. draft, voting, canceled, succeeded, defeated, executed
+    // Jon: since it's possible to update governance parameters while a proposal is in
+    // flight, we can copy over the governance parameters during proposal
+    // creation, such as cooldown period, voting period, and success / failure thresholds
```

Comment:
Sounds good. Does it make more sense to copy the parameters when the proposal is finalized or drafted?

Reply:
Good question... it's probably best to set these parameters when it's finalized and voting is starting, right? But not a huge deal either way

Reply:
Addressed in #15.